phoenix-core/src/main/java/org/apache/phoenix/mapreduce/util/PhoenixMapReduceUtil.java

```diff
@@ -63,6 +67,66 @@ public static void setInput(final Job job, final Class<? extends DBWritable> inp
         PhoenixConfigurationUtil.setSchemaType(configuration, SchemaType.QUERY);
      }
 
+    /**
+     *
+     * @param job
+     * @param inputClass DBWritable class
+     * @param snapshotName The name of a snapshot (of a table) to read from
+     * @param tableName Input table name
+     * @param restoreDir a temporary dir to copy the snapshot files into
+     * @param conditions Condition clause to be added to the WHERE clause. Can be <tt>null</tt> if there are no conditions.
+     * @param fieldNames fields being projected for the SELECT query.
+     */
+    public static void setInput(final Job job, final Class<? extends DBWritable> inputClass, final String snapshotName, String tableName,
```

Comment:
Why wouldn't we want to take the snapshot here instead of passing in the snapshot name?